src/peft/tuners/lora.py

```diff
@@ -233,6 +238,34 @@ def enable_adapter_layers(self):
     def disable_adapter_layers(self):
         self._set_adapter_layers(enabled=False)
 
+    def merge_and_unload(self):
```

Comment:
This would not work when the model is loaded in 8bit. An assertion error for the same would be helpful.

Reply:
Thanks for the pointer! added a check for that